sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthRSA256Service.java

```diff
@@ -68,6 +80,82 @@ public AuthRSA256Service(RsaPublicKeyLoader rsaPublicKeyLoader, long leeway) {
     }
   }
 
+  /**
+   * Validates, that the jwt token issuer is the same as the attribute {@link
+   * LoadedPublicKey#getRequiredIssuer()}. Further it will be validated, that the issuer Url has the
+   * same base as the configured {@link KeySource} source uri, that depends on the implementation.
+   *
+   * @param decodedJWT the decoded json web token.
+   * @param loadedPublicKey the structure with all correlated attributes regarding the key.
+   * @return <code>true</code> if the jwt issuer is the same as the configured requiredIssuer and
+   *     the baseUrl is the same as the {@link KeySource} uri or the configuration attribute
+   *     requiredIssuer is not set. Otherwise <code>false</code> is returned.
+   */
+  private boolean validateIssuer(DecodedJWT decodedJWT, LoadedPublicKey loadedPublicKey) {
+
+    if (StringUtils.isNotBlank(loadedPublicKey.getRequiredIssuer())
+        && (!StringUtils.equals(decodedJWT.getIssuer(), loadedPublicKey.getRequiredIssuer())
+            || !validateIssuerAgainstKeySourceUrl(
+                decodedJWT.getIssuer(), loadedPublicKey.getKeySource()))) {
+      LOG.error(
```

Comment:
I don't think this should be logged as error because you're handling that case properly. I'd rather use WARN or INFO. Compare this article: https://www.section.io/engineering-education/how-to-choose-levels-of-logging/

Reply:
In this case I do not agree. The problem of this line is, when you do receive this log message, this is a ultra critical attack on your environment. It means someone tried to use a valid private key to sign the token, that is the same key as your idp is using. As follow ups task you are enforced to change the keyset for your IDP. This is the reason for choosing the the error level. I'm not sure how we want to set the level for this behavior. I'm fine with warn too. But this is not related to the application state, this log level is just security related. How do we log issues like this one in the commons?

Reply:
@JoergSiebahn : Do you have an opinion on it? With this explanation I would also be fine with ERROR.

Reply:
I see this as a warning because we mitigated the treat correctly. Actually, if someone was able to steal the private key of the IDP, he will most likely set the issuer value correct as well.
In my opinion two cases could be more common here:

- the IDP admin changed the issuer value and did not inform the consumers
- an admin of another configured IDP changed the issuer value with intention or by accident to the same value as the IDP with the mentioned key
 
But [as this log is an error](https://github.com/SDA-SE/sda-dropwizard-commons/pull/1030/files#diff-d7e2444a83694da1cbd5193621217ca2afee1b9091ec0b1e621a1cf35457eb4bR68) too, we could keep error here as well.